drivers/ng_at86rf2xx/ng_at86rf2xx.c

```diff
@@ -115,14 +117,14 @@ void ng_at86rf2xx_reset(ng_at86rf2xx_t *dev)
     cpuid[0] &= ~(0x01);
     cpuid[0] |= 0x02;
     /* copy and set long address */
-    memcpy(&addr_long, cpuid, 8);
-    ng_at86rf2xx_set_addr_long(dev, addr_long);
-    /* now compress the long addr to form the short address */
-    for (int i = 2; i < 8; i++) {
-        cpuid[i & 0x01] ^= cpuid[i];
+#if (CPUID_ID_LEN > 8)
+    for (size_t i = 0; i < 8; i++) {
+        cpuid[i] ^= cpuid[i + (CPUID_ID_LEN - 8)];
     }
```

Comment:
> I don't understand the purpose of lines 112-114, so I left them untouched

They XOR every `cpuid[i]; i >= CPUID_ID_LEN` byte cyclically with the first 8 bytes, so creating an ID from ALL available bytes. Actually very similar to what you do here... 

> Any known case?

Currently not. But why make assumptions here that might not be true in the future and that can easily prevented!